src/ripple/app/paths/PathState.cpp

```diff
@@ -68,7 +67,8 @@ void PathState::reset(STAmount const& in, STAmount const& out)
         << " saOutAct=" << outAct()
         << " saOutReq=" << outReq();
 
-    assert (outAct() < outReq());
+    assert(outAct() < outReq() ||
+        outReq() == STAmount(outReq().issue(), 1u, 0, true));
```

Comment:
Oh, this was the original "what's the -1" comment here, which I hadn't pressed return on!